lib/src/util.dart

```diff
@@ -66,7 +66,22 @@ String longName(Info info, {bool useLibraryUri: false}) {
     // assert(!first || segment is LibraryInfo);
     // (today might not be true for for closure classes).
     if (segment is LibraryInfo) {
-      sb.write(useLibraryUri ? segment.uri : segment.name);
+      if (useLibraryUri && forId && segment.uri.isScheme('file')) {
```

Comment:
let's add  a TODO to remove this when dart2js can be invoked with an app-root custom URI. (what we discussed in the other CL).

Reply:
Do we have an issue for this?